llvm/lib/SYCL/PrepareSYCLOpt.cpp

```diff
@@ -220,13 +222,31 @@ struct PrepareSYCLOpt : public ModulePass {
     }
   }
 
+  struct CheckUnsupportedBuiltinsVisitor
+      : public llvm::InstVisitor<CheckUnsupportedBuiltinsVisitor> {
+    void visitCallInst(CallInst &I) {
+      auto *F = I.getCalledFunction();
+      assert(!(llvm::demangle(std::string(F->getName()))
```

Comment:
The problem is that it is not really an assert we want here but a user error message.

Reply:
OK.
I pushed an alternative where an "error tag" is outputed on stderr if an unsupported builtin is detected.
`sycl-vxx.py` will search for  such error tag and print a user friendly and stop the compilation if it is found.